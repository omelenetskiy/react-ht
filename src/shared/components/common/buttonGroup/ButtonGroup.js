import React, { Fragment } from "react";
import styled from "styled-components";

import Button from "../../styled/button";
import Checkbox from "./checkbox/Checkbox";

import StyledButtonGroup from "../../styled/buttonGroup";
import StyledFilterTitle from "../../styled/filterTitle";

const StyledLabel = styled(Button)`
  min-width: 50px;
  padding: 5px 15px;
  margin: 0 5px;
  transition: all 0.1s ease-in-out;
`;

const ButtonGroup = ({ controls, toggleSearchBy, color, type }) => (
  <StyledButtonGroup type={type}>
    <StyledFilterTitle color={color}>{controls.title}</StyledFilterTitle>

    <div className="filter-buttons">
      {controls.buttons.map(button => (
        <Fragment key={button.title}>
          <Checkbox button={button} toggleSearchBy={toggleSearchBy} />
          <StyledLabel as="label" htmlFor={button.title}>
            {button.title}
          </StyledLabel>
        </Fragment>
      ))}
    </div>
  </StyledButtonGroup>
);

export default ButtonGroup;
