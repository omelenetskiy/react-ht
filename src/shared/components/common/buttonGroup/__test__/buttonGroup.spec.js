import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import 'jest-enzyme';
import 'jest-styled-components';

import ButtonGroup from '../ButtonGroup';

const props = {
  controls: {
    buttons: [
      {
        title: 'title',
      },
      {
        title: 'title',
      },
    ],
  },
  toggleSearchBy: jest.fn(),
  color: 'white',
  type: 'type',
};

describe('ButtonGroup', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<ButtonGroup {...props} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
