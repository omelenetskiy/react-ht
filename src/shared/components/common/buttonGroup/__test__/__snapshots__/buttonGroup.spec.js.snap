// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonGroup should render correctly 1`] = `
<buttonGroup___default
  type="type"
>
  <filterTitle___default
    color="white"
  />
  <div
    className="filter-buttons"
  >
    <Checkbox
      button={
        Object {
          "title": "title",
        }
      }
      toggleSearchBy={[MockFunction]}
    />
    <ButtonGroup__StyledLabel
      as="label"
      htmlFor="title"
    >
      title
    </ButtonGroup__StyledLabel>
    <Checkbox
      button={
        Object {
          "title": "title",
        }
      }
      toggleSearchBy={[MockFunction]}
    />
    <ButtonGroup__StyledLabel
      as="label"
      htmlFor="title"
    >
      title
    </ButtonGroup__StyledLabel>
  </div>
</buttonGroup___default>
`;
